internal/expr/query.go

```diff
@@ -166,26 +170,38 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 			return nil, nil, fmt.Errorf("type %q found in query but not passed to get", typeMember.outerType().Name())
 		}
 		switch tm := typeMember.(type) {
-		case *structField:
-			val := outputVal.Field(tm.index)
-			if !val.CanSet() {
-				return nil, nil, fmt.Errorf("internal error: cannot set field %s of struct %s", tm.name, tm.structType.Name())
+		case primitiveType, structField:
+			switch tm := tm.(type) {
+			case primitiveType:
+				if !outputVal.CanSet() {
+					return nil, nil, fmt.Errorf("internal error: cannot set %s", tm.primitiveType.Name())
+				}
```

Comment:
[Nit]: If this is an internal error I think we should either do not re-check it here or make the code as minimal as possible, we do not need two different messages here to distinguish between structs and primitive types, especially if it means adding another branch in the switch. The last benefit is that we can now simplify the switch to an if statements.

Reply:
Changed to an if statement.

Thinking about it, I'm actually not even sure we need these checks for `CanSet`. It returns false if you have done `reflect.ValueOf` on a none pointer value, but we only accept pointers and then call `Elem` which should be fine.

Though I'm going to leave it in because there might be cases I don't know about.